File: Chapter08/C#/Project/WebServices/3-WCFService/WCFClient.Web/SilverlightResource.svc.cs
using System;
using System.Linq;
using System.Runtime.Serialization;
using System.ServiceModel;
using System.ServiceModel.Activation;
using System.Collections.Generic;
using System.Text;
using System.Data.SqlClient;

namespace WCFClient.Web
{
    [ServiceContract(Namespace = "")]
    [AspNetCompatibilityRequirements(RequirementsMode = AspNetCompatibilityRequirementsMode.Allowed)]
    public class SilverlightResource
    {
        private ResourceInfo Resource;
        public SilverlightResource()
        {
            Resource.Author = "";
            Resource.Title = "";
            Resource.Image = "";
            Resource.URL = "";
            Resource.Type = "";
        }
        [OperationContract]
        public ResourceInfo GetResource(int id)
        {
            GetResourceData(id);
            ResourceInfo localResource = new ResourceInfo();
            localResource.Author = Resource.Author;
            localResource.Title = Resource.Title;
            localResource.Image = Resource.Image;
            localResource.URL = Resource.URL;
            localResource.Type = Resource.Type;
            return localResource;
        }
        private void GetResourceData(int ResourceId)
        {
            // This is where you use your business components. 
            // Method calls on Business components are used to populate the data.

            Resource.Title = "Silverlight 3 How To";
            Resource.Author = "Rajesh Lal";
            Resource.Image = "http://silverlightfun.com/Samples/Chapter-07/AdoData/images/S3HT.jpg";
            Resource.URL = "http://www.amazon.com/Silverlight-3-How-Rajesh-Lal/dp/0672330628";

            try
            {
                string query = "SELECT * FROM Resource where ID=" + ResourceId;

                clsDataAccess mydataAccess = new clsDataAccess();
                mydataAccess.openConnection();
                SqlDataReader mydr = mydataAccess.getData(query);
                if (mydr.HasRows)
                {
                    while (mydr.Read())
                    {
                        Resource.Title = mydr.GetValue(1).ToString();
                        Resource.Author = mydr.GetValue(2).ToString();
                        Resource.Image = mydr.GetValue(3).ToString();
                        Resource.URL = mydr.GetValue(4).ToString();
                        if (Convert.ToInt32(mydr.GetValue(5)) == 1) Resource.Type = "Book";
                        if (Convert.ToInt32(mydr.GetValue(5)) == 2) Resource.Type = "Blog";
                        if (Convert.ToInt32(mydr.GetValue(5)) == 3) Resource.Type = "Tutorial";
                        if (Convert.ToInt32(mydr.GetValue(5)) == 4) Resource.Type = "Video";
              
                    }
                }
                else
                {
                    Resource.Title = "Error: No data for that ID";
                    Resource.Author = "Try other ID's between 1 and 16";
                    Resource.Image = "http://silverlightfun.com/Samples/Chapter-07/AdoData/images/error.jpg";
                    Resource.URL = "http://www.amazon.com/Silverlight-3-How-Rajesh-Lal/dp/0672330628";
                }

                mydr.Close();
                mydataAccess.closeConnection();
            }
            catch (Exception e)
            {
                Resource.Title = "Error: " + e.Message;
                Resource.Author = "Occured while connecting to the Database ";
                Resource.Image = "http://silverlightfun.com/Samples/Chapter-07/AdoData/images/error.jpg";
                Resource.URL = "http://www.amazon.com/Silverlight-3-How-Rajesh-Lal/dp/0672330628";
            }
        }
        public struct ResourceInfo
        {
            public string Author { get; set; }
            public string Title { get; set; }
            public string Image { get; set; }
            public string URL { get; set; }
            public string Type { get; set; }
        }
        // Add more operations here and mark them with [OperationContract]
    }
}

File: Chapter08/VB/VBProject_08/WebServices/3-WCFService/WCFClient.Web/SilverlightResource.svc.vb
Imports System
Imports System.Linq
Imports System.Runtime.Serialization
Imports System.ServiceModel
Imports System.ServiceModel.Activation
Imports System.Collections.Generic
Imports System.Text
Imports System.Data.SqlClient

' Namespace WCFClient.Web

    <ServiceContract(Namespace := "")>
    <AspNetCompatibilityRequirements(RequirementsMode := AspNetCompatibilityRequirementsMode.Allowed)>
    Public Class SilverlightResource

    Private oResource As ResourceInfo

        Public Sub New()

        oResource.Author = ""
        oResource.Title = ""
        oResource.Image = ""
        oResource.URL = ""
        oResource.Type = ""
        End Sub '   New

    <OperationContract()>
    Public Function GetResource(id As Integer) As ResourceInfo

        GetResourceData(id)

        Dim localResource As ResourceInfo = New ResourceInfo()

        localResource.Author = oResource.Author
        localResource.Title = oResource.Title
        localResource.Image = oResource.Image
        localResource.URL = oResource.URL
        localResource.Type = oResource.Type
        Return localResource
    End Function  '   GetResource

    Private Sub GetResourceData(ResourceId As Integer)

        '  This is where you use your business components.
        '  Method calls on Business components are used to populate the data.

        oResource.Title = "Silverlight 3 How To"
        oResource.Author = "Rajesh Lal"
        oResource.Image = "http://silverlightfun.com/Samples/Chapter-07/AdoData/images/S3HT.jpg"
        oResource.URL = "http://www.amazon.com/Silverlight-3-How-Rajesh-Lal/dp/0672330628"

        Try

            Dim query As String = "SELECT * FROM Resource where ID=" + CStr(ResourceId)

            Dim mydataAccess As clsDataAccess = New clsDataAccess()

            mydataAccess.openConnection()

            Dim mydr As SqlDataReader = mydataAccess.getData(query)

            If (mydr.HasRows) Then

                While (mydr.Read())

                    oResource.Title = mydr.GetValue(1).ToString()
                    oResource.Author = mydr.GetValue(2).ToString()
                    oResource.Image = mydr.GetValue(3).ToString()
                    oResource.URL = mydr.GetValue(4).ToString()

                    Select Case (Convert.ToInt32(mydr.GetValue(5)))
                        Case 1
                            oResource.Type = "Book"
                        Case 2
                            oResource.Type = "Blog"
                        Case 3
                            oResource.Type = "Tutorial"
                        Case 4
                            oResource.Type = "Video"
                    End Select
                End While   '
            Else
                oResource.Title = "Error: No data for that ID"
                oResource.Author = "Try other ID's between 1 and 16"
                oResource.Image = "http://silverlightfun.com/Samples/Chapter-07/AdoData/images/error.jpg"
                oResource.URL = "http://www.amazon.com/Silverlight-3-How-Rajesh-Lal/dp/0672330628"
            End If

            mydr.Close()
            mydataAccess.closeConnection()
        Catch e As Exception
            oResource.Title = "Error: " + e.Message
            oResource.Author = "Occured while connecting to the Database "
            oResource.Image = "http://silverlightfun.com/Samples/Chapter-07/AdoData/images/error.jpg"
            oResource.URL = "http://www.amazon.com/Silverlight-3-How-Rajesh-Lal/dp/0672330628"
        End Try
    End Sub '   GetResourceData

    Public Structure ResourceInfo

        Public Property Author As String
        Public Property Title As String
        Public Property Image As String
        Public Property URL As String
        Public Property Type As String
    End Structure
    '  Add more operations here and mark them with (OperationContract)
End Class   '   SilverlightResource
' End Namespace
' ..\Project_08\WebServices\3-WCFService\WCFClient.Web\SilverlightResource.svc.cs
